27-paddlefish/technical_specification.md

```diff
@@ -0,0 +1,47 @@
+# Implementation of encryption and parallel file part processing in uploads and downloads for ghga-connector (Paddlefish)
+
+**Epic Type:** Implementation Epic
+
+**Attention: Please do not put any confidential content here.**
+
+## Scope:
+
+This epic covers two features to be implemented in the ghga-connector:
+
+1. Crypt4GH compliant encryption of files to be uploaded
+2. Parallelization of file part uploads/downloads
+
+
+## Implementation Details:
+
+### Encryption
+
+Crypt4GH encryption needs the user's private key and the GHGA public key.
+For now, both these keys should be provided by adding configuration options that expect the respective paths to the key files.
```

Comment:
Not entirely clear, what you mean by "configuration options", but we want to handle those keys differently.
The users private/public keypair should be given through CLI options, while the GHGA public key, which we don't know where it will come from yet, will be passed on through our config file.

Reply:
Fair point, makes sense to pass the GHGA public key in a different way